vignettes/articles/scripting-tasks-config.Rmd

```diff
@@ -0,0 +1,387 @@
+---
+title: "Scripting task configuration files"
+---
+
+```{r, setup, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+## Introduction
+
+The `tasks.json` configuration file is a the source of truth for validating a
+hub and is also one of the most complex aspects of a hub. This configuration
+file is a formal representation of an official collaborative forecasting
+challenge that allows direct and unambiguous evaluation and comparison of model
+outputs. It allows hub administrators to set a clear standard for model
+submissions.
```

Comment:
I think the phrase "source of truth" could be a little confusing because of our/the community's historical use of "truth" for target data. I also think it's nice to lead with what the config achieves. Hence the suggested revision below:
```suggestion
The `tasks.json` configuration file is a formal representation of a collaborative forecasting
challenge. This configuration file sets a clear standard for model outputs in a hub,
which enables validations of model output submissions to ensure that they are correctly
formatted. Configuration settings in the tasks configuration file also contain structured
information about modeling tasks that can be used for downstream applications like
evaluations and visualizations of model outputs.
```